segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -229,7 +229,8 @@ private MetadataStore createMetadataStore() {
             ChunkedSegmentStorage chunkedStorage = (ChunkedSegmentStorage) this.storage;
             val snapshotInfoStore = getStorageSnapshotInfoStore();
             // Bootstrap
-            return chunkedStorage.bootstrap(snapshotInfoStore);
+            StorageEventProcessor eventProcessor = new StorageEventProcessor(this.metadata.getContainerId(), this.containerEventProcessor, chunkedStorage);
```

Comment:
See my comment in `ChunkedSegmentStorage` below regarding this instance.

Furthermore you need to ensure that if `.bootstrap` fails, whether sync or async, that you close `eventProcessor` here; otherwise this is another source of leaks.

Reply:
Fixed. Closing it with ChunkedSegmentStorage::close